### DISCUSSION TITLE
Serializer and static property

### DISCUSSION THREAD

**Discussion by etshy** [symfony/symfony#59011] [Q&A]
I recently got issue with serializing an object that have a public static property, getting error "Accessing static property XX::$yy as non static", using Symfony 6.4.12.
I'm not serializing the object myself though it's going to `joliecode/elastically` `Indexer::index` method, so I don't have direct control on the serializer.
Are static properties not "compatible" with Symfony Serializer ?

**Comment by stof** [comment]
A bug related to serializing classes with static properties has been fixed in the 6.4.13 release. Maybe your error is related to that bug.
Make sure you update to the latest version of Symfony first as we fix issues regularly (6.4.12 is late by 3 releases at the time of writing, as the latest 6.4.x release is currently 6.4.15)

**Reply by etshy** [reply]
Well I'd likk to keep dependencies up to date, but it's for a work project and leads doesn't want to update everything too often (I guess they have reasons to do that).

Though seeing on `PropertyAccessor` on the 6.4 branch, there is no check if property is static before accessing it https://github.com/symfony/symfony/blob/2982bb7ffc7adb1a3e5ab290c2b5f303d8e585d1/src/Symfony/Component/PropertyAccess/PropertyAccessor.php#L442-L442
and on `ObjectNormalizer` : https://github.com/symfony/symfony/blob/2982bb7ffc7adb1a3e5ab290c2b5f303d8e585d1/src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php#L136

it seems static properties are just ignored now ? (same for static method).
I tried it quickly and my static property is indeed not serialized (into json).

Is that intended ?

On `PropertyAccessor` there is a `$access` variable with informations if the property is static (among other things).
Shouldn't static properties be included during serializing and `PropertyAccessor` check to access the property according to its `readInfo` (static, visibility, etc.) ?
